src/server.c

```diff
@@ -3410,7 +3470,8 @@ void call(client *c, int flags) {
     if (flags & CMD_CALL_PROPAGATE &&
         (c->flags & CLIENT_PREVENT_PROP) != CLIENT_PREVENT_PROP &&
         c->cmd->proc != execCommand &&
-        !(c->cmd->flags & CMD_MODULE))
+        !(c->cmd->flags & CMD_MODULE) &&
+        ((c->cmd->flags & CMD_WRITE) || (c->cmd->flags & CMD_MAY_REPLICATE)))
```

Comment:
why?

Reply:
As mentioned on the top comment:

> key miss event:
On key miss event, if a module performed some write command on the event (using RM_Call), the dirty counter would have increase and the read command that cause the key miss event would have been replicated to the replication and aof. To avoid this, we perform the replication only on WRITE or MAY_REPLICATE commands. This will prevent replicating read commands even if the dirty counter was increased. Test was added to verify this scenario.

Added a comment on the code itself as well.

Reply:
but what about lazy-expire? how does that work? imagine a GET that causes the propagation of DEL... from what i can see from the code they would both propagate (which is not the best) but now after your change none of them would propagate (which is worse). please CMIIW and add a test if I'm not

Reply:
I believe only the `get` will not be propagated, let me write a test for it.

Reply:
Pushed another commit with the test.